packages/driver/src/cy/commands/navigation.coffee

```diff
@@ -304,15 +304,14 @@ module.exports = (Commands, Cypress, cy, state, config) ->
         when not resp.isOkStatusCode
           err = new Error
           err.gotResponse = true
-          _.extend(err, resp)
-
+          _.extend(err, resp, { originalUrl: url});
```

Comment:
this is just to help with debugging, right?

Reply:
No see this comment
> When do we expect the error message to be `/foo.html` vs `http://localhost:3500/whatever`?
> I changed this code because with the query param, instead of the error throwing `404 from http://localhost:3500?q=foo`
> it would throw `404 from http://localhost:3500/?q=foo` even though the request was sent without the slash. Error message, I'm referring to is the "visit.loading_type_failed" message

For some reason when a request to `http://localhost:3500?q=foo would 404, instead of displaying the error message as `failed to load http://localhost:3500?q=foo` it would display it as `failed to load http://localhost:3500/?q=foo`

Reply:
Not sure the best way to handle this to be honest. I could add some logic on which url to use but that seems more complicated than it should be. Not sure how big of deal it is that the error displays the wrong url? @flotwig

Reply:
I think that `http://localhost:3500/?q=foo` is the correct URL to show. Chrome and Node will both proxy `http://localhost:3500?q=foo` like this:

```
GET /?q=foo HTTP/1.1
Host: localhost:3500
```

Since the request is actually made with the `/` prefixed, I think it's fine for the error to reflect that.

Reply:
Right, gotcha. I didn't think to actual check the request that was being made